src/main/java/org/elasticsearch/index/query/CommonTermsQueryParser.java

```diff
@@ -160,64 +141,31 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
         if (value == null) {
             throw new QueryParsingException(parseContext, "No text specified for text query");
         }
-        String field;
-        FieldMapper mapper = parseContext.fieldMapper(fieldName);
-        if (mapper != null) {
-            field = mapper.names().indexName();
-        } else {
-            field = fieldName;
+        CommonTermsQueryBuilder commonTermsQuery = new CommonTermsQueryBuilder(fieldName, value.toString())
```

Comment:
`value.toString` here looks fishy. Why do we parse an object (parser.objectText()) if we then convert it to String when we create the actual query? does this query support numeric values or not?